### PR TITLE
Add SEDE dataset

### PR DESCRIPTION
**Pull request by Hazoom**
This PR adds the SEDE dataset for the task of realistic Text-to-SQL, following the instructions of how to add a database and a dataset card.

Please see our paper for more details: https://arxiv.org/abs/2106.05006

### CODE REVIEW

File: datasets/sede/README.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+annotations_creators:
+- no-annotation
+language_creators:
+- found
+languages:
+- en
+licenses:
+- apache-2.0
+multilinguality:
+- monolingual
```
Comment (albertvillanova) [inline datasets/sede/README.md:12]: Could you please also add the tag with Papers With Code ID?
```suggestion
- monolingual
paperswithcode_id: sede
```

File: datasets/sede/sede.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            description="SEDE Dataset: A Naturally-Occurring Dataset Based on Stack Exchange Data.",
+        ),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "QuerySetId": datasets.Value("string"),
```
Comment (albertvillanova) [inline datasets/sede/sede.py:66]: Isn't this field integer data type? If so, this would reduce the storage size.

File: datasets/sede/sede.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"data_filepath": val_path}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"data_filepath": test_path}),
+        ]
+
+    def _generate_examples(self, data_filepath):
+        """Generate SEDE examples."""
+        logger.info("generating examples from = %s", data_filepath)
+        with open(data_filepath, encoding="utf-8") as f:
+            content = list(f)
+            for idx, sample_str in enumerate(content):
```
Comment (albertvillanova) [inline datasets/sede/sede.py:95]: I think this code line is not necessary. If removed, you will avoid loading all the dataset in memory (and loading it incrementally).
```suggestion
            for idx, sample_str in enumerate(f):
```

File: datasets/sede/sede.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                sample_json = json.loads(sample_str)
+                yield idx, {
+                    "QuerySetId": str(sample_json["QuerySetId"]),
```
Comment (albertvillanova) [inline datasets/sede/sede.py:98]: If you set this field data type as integer, you will not need to cast it to string:
```suggestion
                    "QuerySetId": sample_json["QuerySetId"],
```

File: datasets/sede/sede.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                    "Title": sample_json["Title"],
+                    "Description": sample_json["Description"],
+                    "QueryBody": sample_json["QueryBody"],
+                    "CreationDate": sample_json["CreationDate"],
+                    "validated": sample_json["validated"],
+                }
```
Comment (albertvillanova) [inline datasets/sede/sede.py:104]: And then, if the `sample_json` contains only the required fields, you could yield it directly:
```suggestion
                yield idx, sample_json
```

**Review by albertvillanova — CHANGES_REQUESTED** [review]
Thanks @Hazoom for the addition of this dataset. You did a great job!! :)

Below just some comments and minor requested changes.

**Comment by Hazoom** [comment]
Thanks @albertvillanova for your great suggestions! I just pushed a new commit with the necessary fixes.

File: datasets/sede/sede.py (hunk 2 above)
Comment (Hazoom) [reply datasets/sede/sede.py]: Yes, I just commit a fix. Thanks for noticing.

**Comment by albertvillanova** [comment]
Hi @Hazoom,

You were right: the non-passing test had nothing to do with this PR.

Unfortunately, you did a git rebase (instead of a git merge), which is not recommended once you have already open a pull request because you mess up your pull request history. You can see that your pull request now contains:
- your commits repeated two times
- and commits which are not yours from the master branch

If you would like to clean your pull request, please make:
```
git reset --hard 587b93a
git fetch upstream master
git merge upstream/master
git push --force origin sede
```

**Comment by Hazoom** [comment]
> Hi @Hazoom,
> 
> You were right: the non-passing test had nothing to do with this PR.
> 
> Unfortunately, you did a git rebase (instead of a git merge), which is not recommended once you have already opened a pull request because you mess up your pull request history. You can see that your pull request now contains:
> 
> * your commits repeated two times
> * and commits which are not yours from the master branch
> 
> If you would like to clean your pull request, please make:
> 
> ```
> git reset --hard 587b93a
> git fetch upstream master
> git merge upstream/master
> git push --force origin sede
> ```

Thanks @albertvillanova

**Comment by albertvillanova** [comment]
Nice! Just one final request before approving your pull request:

As you have updated the "QuerySetId" field data type, the size of the dataset is smaller now. You should regenerate the metadata again. Please run:
```
rm datasets/sede/dataset_infos.json
datasets-cli test datasets/sede --save_infos --all_configs
```

**Review by albertvillanova — CHANGES_REQUESTED** [review]
Nice! Just one final request before approving your pull request:

As you have updated the "QuerySetId" field data type, the size of the dataset is smaller now. You should regenerate the metadata. Please run:
```
rm datasets/sede/dataset_infos.json
datasets-cli test datasets/sede --save_infos --all_configs
```

**Comment by Hazoom** [comment]
> Nice! Just one final request before approving your pull request:
> 
> As you have updated the "QuerySetId" field data type, the size of the dataset is smaller now. You should regenerate the metadata. Please run:
> 
> ```
> rm datasets/sede/dataset_infos.json
> datasets-cli test datasets/sede --save_infos --all_configs
> ```

@albertvillanova Good catch, just fixed it.

**Review by albertvillanova — APPROVED** [review]
Awesome!!! Thanks a lot for your contribution!

**Review by lhoestq — APPROVED** [review]
Thank you for adding this one :)

**Pull request merged**